x-pack/plugins/security_solution/public/common/components/discover_in_timeline/provider.tsx

```diff
@@ -24,14 +24,16 @@ export const DiscoverInTimelineContextProvider: FC<DiscoverInTimelineContextProv
     discoverStateContainer.current = stateContainer;
   }, []);
 
+  const discoverInTimelineContext = useMemo(() => {
+    return {
+      discoverStateContainer,
+      setDiscoverStateContainer,
+      ...actions,
+    };
+  }, [actions, setDiscoverStateContainer]);
```

Comment:
Nice! I think we may need to either add `discoverStateContainer.current` to the dependency array, though I'm not sure if the old `.current` ref is retained or if it's properly updated even with the object memoized 🤔